python/lsst/ts/adamSensors/adamSensorsCSC.py

```diff
@@ -1,48 +1,155 @@
 from lsst.ts import salobj
 from lsst.ts.adamSensors.model import AdamModel
+from numpy import poly1d
 import pathlib
 import asyncio
+import concurrent
 
 
 class AdamCSC(salobj.ConfigurableCsc):
     """
     CSC for simple sensors connected to an ADAM controller
     """
 
-    def __init__(self, config_dir=None, initial_state=salobj.State.STANDBY, initial_simulation_mode=0):
-        schema_path = pathlib.Path(__file__).resolve().parents[4].joinpath("schema", "AdamSensors.yaml")
-        super().__init__("AdamSensors", index=0, schema_path=schema_path, config_dir=config_dir,
-                         initial_state=initial_state, initial_simulation_mode=initial_simulation_mode)
+    def __init__(
+        self, config_dir=None, initial_state=salobj.State.STANDBY, simulation_mode=0
+    ):
+        schema_path = (
+            pathlib.Path(__file__)
+            .resolve()
+            .parents[4]
+            .joinpath("schema", "AdamSensors.yaml")
+        )
+        super().__init__(
+            "AdamSensors",
+            index=0,
+            schema_path=schema_path,
+            config_dir=config_dir,
+            initial_state=initial_state,
+            simulation_mode=simulation_mode,
+        )
 
-        self.connected = False
         self.adam = None
         self.config = None
+        self.start_timeout = 10
+        self.valid_simulation_modes = [0, 1]
 
         # setup asyncio tasks for the loop
         done_task = asyncio.Future()
         done_task.set_result(None)
         self.telemetryLoopTask = done_task
 
-    async def begin_start(self, id_data):
-        await super().begin_start(id_data)
-        self.adam = AdamModel(self.log, simulation_mode=True)
-        await self.adam.connect(self.config.ip, self.config.port)
+    async def begin_start(self, data):
+        self.cmd_start.ack_in_progress(data, timeout=self.start_timeout)
+        await super().begin_start(data)
+        self.adam = AdamModel(self.log, simulation_mode=self.simulation_mode)
+        self.telemetryLoopTask = asyncio.create_task(self.telemetry_loop())
+        await self.telemetry_loop()
```

Comment:
Now you are doing `telemetry_loop` twice. One in the background and another one here. Remove this line.